eras/conway/impl/cddl/lib/Cardano/Ledger/Conway/HuddleSpec.hs

```diff
@@ -0,0 +1,1256 @@
+conwayWithdrawalsRule p =
+  "withdrawals"
+    =:= mp
+      [ 1
+          <+ asKey (huddleRule @"reward_account" p)
+          ==> huddleRule @"coin" p
+      ]
+
+conwayRedeemerTag :: Rule
+conwayRedeemerTag =
+  comment
+    [str|0: spend
+        |1: mint
+        |2: cert
+        |3: reward
+        |4: voting
+        |5: proposing
+        |]
+    $ "redeemer_tag"
+      =:= (0 :: Integer)
+      ... (5 :: Integer)
+
+conwayRedeemer ::
+  forall era.
+  ( HuddleRule "redeemer_tag" era
+  , HuddleRule "plutus_data" era
+  , HuddleRule "ex_units" era
+  ) =>
+  Proxy era ->
+  Rule
+conwayRedeemer p =
+  "redeemer"
+    =:= arr
+      [ "tag" ==> huddleRule @"redeemer_tag" p
+      , "index" ==> (VUInt `sized` (4 :: Word64))
+      , "data" ==> huddleRule @"plutus_data" p
+      , "ex_units" ==> huddleRule @"ex_units" p
+      ]
+
+instance HuddleRule "min_int64" ConwayEra where
+  huddleRule _ = minInt64Rule
+
+instance HuddleRule "max_int64" ConwayEra where
+  huddleRule _ = maxInt64Rule
+
+instance HuddleRule "int64" ConwayEra where
+  huddleRule = int64Rule @ConwayEra
+
+instance HuddleRule "bounded_bytes" ConwayEra where
+  huddleRule _ = boundedBytesRule
+
+instance HuddleRule "distinct_bytes" ConwayEra where
+  huddleRule _ = distinctBytesRule
+
+instance HuddleRule "big_uint" ConwayEra where
+  huddleRule = bigUintRule
+
+instance HuddleRule "big_nint" ConwayEra where
+  huddleRule = bigNintRule
+
+instance HuddleRule "big_int" ConwayEra where
+  huddleRule = bigIntRule
+
+instance HuddleRule "network_id" ConwayEra where
+  huddleRule _ = networkIdRule
+
+instance HuddleRule "dns_name" ConwayEra where
+  huddleRule _ = "dns_name" =:= VText `sized` (0 :: Word64, 128 :: Word64)
+
+instance HuddleRule "url" ConwayEra where
+  huddleRule _ = "url" =:= VText `sized` (0 :: Word64, 128 :: Word64)
+
+instance HuddleRule "major_protocol_version" ConwayEra where
+  huddleRule = majorProtocolVersionRule @ConwayEra
+
+instance HuddleRule "genesis_hash" ConwayEra where
+  huddleRule = genesisHashRule @ConwayEra
+
+instance HuddleRule "genesis_delegate_hash" ConwayEra where
+  huddleRule = genesisDelegateHashRule @ConwayEra
+
+instance HuddleRule "transaction_id" ConwayEra where
+  huddleRule = transactionIdRule @ConwayEra
+
+instance HuddleRule "vkeywitness" ConwayEra where
+  huddleRule = vkeywitnessRule @ConwayEra
+
+instance HuddleRule "bootstrap_witness" ConwayEra where
+  huddleRule = bootstrapWitnessRule @ConwayEra
+
+instance HuddleRule "ex_units" ConwayEra where
+  huddleRule _ = exUnitsRule
+
+instance HuddleRule "positive_interval" ConwayEra where
+  huddleRule = positiveIntervalRule
+
+instance HuddleRule "vote" ConwayEra where
+  huddleRule _ = "vote" =:= (0 :: Integer) ... (2 :: Integer)
+
+instance HuddleRule "asset_name" ConwayEra where
+  huddleRule _ = "asset_name" =:= VBytes `sized` (0 :: Word64, 32 :: Word64)
+
+instance HuddleRule "plutus_data" ConwayEra where
+  huddleRule = plutusDataRule
```

Comment:
This instance can go to `cardano-ledger-alonzo`, since `plutus_dat` can never change, otherwise it would yield a whole lot of funds locked forever.
```suggestion
instance HuddleRule "plutus_data" era where
  huddleRule = plutusDataRule
```